ogr/ogrsf_frmts/miramon/mm_gdal_functions.c

```diff
@@ -0,0 +1,2801 @@
+
+    i_camp = 0;
+
+    MM_FillFieldDB_XP(
+        bd_xp->pField + i_camp, szMMNomCampIdGraficDefecte,
+        szInternalGraphicIdentifierEng, szInternalGraphicIdentifierCat,
+        szInternalGraphicIdentifierSpa, 'N', MM_MAX_AMPLADA_CAMP_N_DBF, 0);
+    bd_xp->IdGraficField = 0;
+    (bd_xp->pField + i_camp)->GeoTopoTypeField = (MM_BYTE)MM_CAMP_ES_ID_GRAFIC;
+    i_camp++;
+
+    MM_FillFieldDB_XP(bd_xp->pField + i_camp, szMMNomCampArcsANodeDefecte,
+                      szNumberOfArcsToNodeEng, szNumberOfArcsToNodeCat,
+                      szNumberOfArcsToNodeSpa, 'N', MM_MAX_AMPLADA_CAMP_N_DBF,
+                      0);
+    (bd_xp->pField + i_camp)->GeoTopoTypeField = (MM_BYTE)MM_CAMP_ES_ARCS_A_NOD;
+    i_camp++;
+
+    MM_FillFieldDB_XP(bd_xp->pField + i_camp, szMMNomCampTipusNodeDefecte,
+                      szNodeTypeEng, szNodeTypeCat, szNodeTypeSpa, 'N', 1, 0);
+    (bd_xp->pField + i_camp)->GeoTopoTypeField = (MM_BYTE)MM_CAMP_ES_TIPUS_NODE;
+    i_camp++;
+
+    return i_camp;
+}
+
+size_t MM_DefineFirstPointFieldsDB_XP(struct MM_DATA_BASE_XP *bd_xp)
+{
+    size_t i_camp = 0;
+
+    MM_FillFieldDB_XP(
+        bd_xp->pField + i_camp, szMMNomCampIdGraficDefecte,
+        szInternalGraphicIdentifierEng, szInternalGraphicIdentifierCat,
+        szInternalGraphicIdentifierSpa, 'N', MM_MAX_AMPLADA_CAMP_N_DBF, 0);
+    bd_xp->IdGraficField = 0;
+    (bd_xp->pField + i_camp)->GeoTopoTypeField = (MM_BYTE)MM_CAMP_ES_ID_GRAFIC;
+    i_camp++;
+
+    return i_camp;
+}
+
+static int MM_SprintfDoubleWidth(char *cadena, int amplada, int n_decimals,
+                                 double valor_double,
+                                 MM_BOOLEAN *Error_sprintf_n_decimals)
+{
+#define VALOR_LIMIT_IMPRIMIR_EN_FORMAT_E 1E+17
+#define VALOR_MASSA_PETIT_PER_IMPRIMIR_f 1E-17
+    char cadena_treball[MM_CARACTERS_DOUBLE + 1];
+    int retorn_printf;
+
+    if (MM_IsNANDouble(valor_double))
+    {
+        if (amplada < 3)
+        {
+            *cadena = *MM_EmptyString;
+            return EOF;
+        }
+        return sprintf(cadena, "NAN");
+    }
+    if (MM_IsDoubleInfinit(valor_double))
+    {
+        if (amplada < 3)
+        {
+            *cadena = *MM_EmptyString;
+            return EOF;
+        }
+        return sprintf(cadena, "INF");
+    }
+
+    *Error_sprintf_n_decimals = FALSE;
+    if (valor_double == 0)
+    {
+        retorn_printf =
+            sprintf(cadena_treball, "%*.*f", amplada, n_decimals, valor_double);
+        if (retorn_printf == EOF)
+        {
+            *cadena = *MM_EmptyString;
+            return retorn_printf;
+        }
+
+        if (retorn_printf > amplada)
+        {
+            int escurcament = retorn_printf - amplada;
+            if (escurcament > n_decimals)
+            {
+                *cadena = *MM_EmptyString;
+                return EOF;
+            }
+            *Error_sprintf_n_decimals = TRUE;
+            n_decimals = n_decimals - escurcament;
+            retorn_printf =
```

Comment:
same comment as above

Reply:
There are three. After the first one I saw that a cast is missing. So:
if (retorn_printf >= **(int)**sizeof(cadena_treball))

Reply:
There are 4, not 3.